c10/core/LinalgBackend.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <c10/util/Exception.h>
+
+#include <ostream>
+#include <string>
+
+namespace c10 {
+
+enum class LinalgBackend : int8_t { Default, Cusolver, Magma };
```

Comment:
@ngimel suggests that c10 shouldn't need to call this, so this file should be in ATen near where the code for the cudnn enabling logic lives